client/my-sites/plans-features-main/components/free-free-dialog.tsx

```diff
@@ -0,0 +1,290 @@
+import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
+import { getPlanRawPrice } from 'calypso/state/plans/selectors';
+import { DialogContainer } from './free-plan-paid-domain-dialog';
+import { LoadingPlaceHolder } from './loading-placeholder';
+
+export const Heading = styled.div`
+	font-family: Recoleta;
+	color: var( --studio-gray-100 );
+	font-size: 22px;
+	line-height: 26px;
+	letter-spacing: 0.38px;
+	margin-bottom: 12px;
+	@media ( min-width: 780px ) {
+		font-size: 32px;
+		line-height: 40px;
+		letter-spacing: -0.32px;
+	}
+`;
+
+const List = styled.ul`
+	list-style: none;
+	margin: 20px 0 12px;
+	font-weight: 600;
+	font-size: 14px;
+`;
+const ListItem = styled.li`
+	display: flex;
+	& div:first-of-type {
+		margin: 0 8px 0 8px;
+	}
+`;
+const ButtonRow = styled.div`
+	display: flex;
+	justify-content: flex-start;
+	margin: 16px 0;
+`;
+
+const TextBox = styled.div< { fontSize?: number; bold?: boolean; color?: 'gray' } >`
+	font-size: ${ ( { fontSize } ) => fontSize || 14 }px;
+	font-weight: ${ ( { bold } ) => ( bold ? 600 : 400 ) };
+	line-height: 20px;
+	margin-bottom: 8px;
+	color: ${ ( { color } ) => {
+		if ( color === 'gray' ) {
+			return 'var(--studio-gray-50)';
+		}
+		return 'var(--color-text)';
+	} };
+`;
+
+export const StyledButton = styled( Button )`
+	padding: 10px 24px;
+	border-radius: 4px;
+	font-weight: 500;
+	font-size: 14px;
+	line-height: 20px;
+	flex: 1;
+	&.is-borderless {
+		text-decoration: underline;
+		border: none;
+		font-weight: 600;
+		padding: 0;
+		color: ${ ( { color } ) => {
+			if ( color === 'gray' ) {
+				return 'var(--studio-gray-50)';
+			}
+			return 'var(--color-text)';
+		} };
+	}
+
+	&.is-primary,
+	&.is-primary.is-busy,
+	&.is-primary:hover,
+	&.is-primary:focus {
+		background-color: var( --studio-blue-50 );
+		border: unset;
+	}
+
+	&:hover {
+		opacity: 0.85;
+		transition: 0.7s;
+	}
+
+	width: 100%;
+	@media ( min-width: 780px ) {
+		max-width: 260px;
+		width: unset;
+	}
+`;
+
+const CrossIcon = styled( Gridicon )`
+	color: #e53e3e;
+`;
+
+const LoadingPlaceHolderText = styled( LoadingPlaceHolder )`
+	width: 80px;
+	display: inline-block;
+	border-radius: 0;
+`;
+
+function LazyDisplayText( {
+	displayText = '',
+	isLoading,
+}: {
+	displayText?: TranslateResult;
+	isLoading: boolean;
+} ) {
+	return isLoading || ! displayText ? <LoadingPlaceHolderText /> : <>{ displayText }</>;
+}
+
+export function FreeFreeDialog( {
+	freeSubdomain,
+	onFreePlanSelected,
+	onPlanSelected,
+	onClose,
+}: {
+	freeSubdomain: string;
+	onClose: () => void;
+	onFreePlanSelected: () => void;
+	onPlanSelected: () => void;
+} ) {
+	const translate = useTranslate();
+	const currencyCode = useSelector( getCurrentUserCurrencyCode ) ?? 'USD';
+
+	const planConstantObj = applyTestFiltersToPlansList( PLAN_PERSONAL, undefined );
+	const planProductId = planConstantObj.getProductId();
+	const rawPrice = useSelector( ( state ) => getPlanRawPrice( state, planProductId, true ) );
```

Comment:
Using the `rawPrice` would show an incorrect price whenever currency-specific discounts are active.
<img width="1218" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/22e97ea1-5adb-419a-b13f-77c59eaf9937">

You can use the `usePlanPrices` hook instead, here's an example:
https://github.com/Automattic/wp-calypso/blob/bc7eafb72b5a966fb4a302b62f2d20c6a348c5ac/client/my-sites/plans-features-main/components/free-plan-paid-domain-dialog.tsx#L135